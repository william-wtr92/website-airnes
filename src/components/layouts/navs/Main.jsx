import { useRouter } from "next/router"
import Admin from "./Admin"
import Users from "@/components/layouts/navs/Users"

const Main = () => {
  const router = useRouter()

  return !router.pathname.startsWith("/admin") ? (
    <Users/>
    ) : (
    <Admin />
  )
}

export default Main
