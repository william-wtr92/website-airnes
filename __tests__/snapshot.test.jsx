import { render } from "@testing-library/react"
import Main from "@/pages/index"

describe("Main page", () => {
  it("renders correctly", () => {
    const mockProps = {
      carousel: [],
      categories: [],
      selectedProduct: [],
      products: [],
    }

    const tree = render(<Main {...mockProps} />)
    expect(tree).toMatchSnapshot()
  })
})
