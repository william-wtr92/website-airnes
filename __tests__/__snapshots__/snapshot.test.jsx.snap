// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main>
    <div
      class="flex flex-col gap-6 justify-center items-center h-72 lg:h-96"
    >
      <div
        class="animate-spin w-8 h-8 border-t-2 border-transparent border-primary rounded-full"
      />
      <p
        class="font-bold"
      >
        Loading ...
      </p>
    </div>
    <div
      class="flex flex-col gap-8"
    >
      <div
        class="text-center text-[13px] font-bold lg:py-6 lg:text-xl"
      >
        <p>
          VENANT DES HAUTES TERRES D’ECOSSE
          <br />
          NOS MEUBLES SONT IMMORTELS
        </p>
      </div>
      <div
        class="flex flex-wrap justify-center"
      >
        <div
          class="flex flex-wrap justify-center lg:justify-between"
        >
          <div
            class="flex flex-col gap-6 justify-center items-center h-72 lg:h-96"
          >
            <div
              class="animate-spin w-8 h-8 border-t-2 border-transparent border-primary rounded-full"
            />
            <p
              class="font-bold"
            >
              Loading ...
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-8"
    >
      <div
        class="text-center text-[13px] font-bold lg:py-6 lg:text-xl"
      >
        <p>
          PRODUITS POPULAIRES
        </p>
      </div>
      <div
        class="flex flex-wrap justify-center"
      >
        <div
          class="flex flex-wrap justify-center lg:justify-between"
        >
          <div
            class="flex flex-col gap-6 justify-center items-center h-72 lg:h-96"
          >
            <div
              class="animate-spin w-8 h-8 border-t-2 border-transparent border-primary rounded-full"
            />
            <p
              class="font-bold"
            >
              Loading ...
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex justify-center py-10"
    >
      <div
        class="flex flex-col items-center gap-10 w-[80%]"
      >
        <h3
          class="uppercase font-bold text-xl tracking-widest"
        >
          Promotions
        </h3>
        <div
          class="overflow-x-auto scrollbar w-full flex gap-10 p-4 justify-center"
        >
          <div
            class="flex-none w-full md:w-1/2 lg:w-1/3"
          >
            <a
              class="no-underline"
              href="/products/1/product"
            >
              <img
                alt="Product 1"
                class="border-2 border-black w-full h-60 object-cover"
                data-nimg="1"
                decoding="async"
                height="500"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=1080&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage1.jpg&w=1080&q=75 2x"
                style="color: transparent;"
                width="500"
              />
              <div
                class="flex flex-col font-extrabold uppercase p-1"
              >
                <h1>
                  Product 1
                </h1>
                <div
                  class="text-right"
                >
                  <p
                    class="line-through"
                  >
                    100
                     €
                  </p>
                  <p
                    class="text-red-600 text-lg"
                  >
                    10
                     €
                  </p>
                </div>
              </div>
            </a>
          </div>
          <div
            class="flex-none w-full md:w-1/2 lg:w-1/3"
          >
            <a
              class="no-underline"
              href="/products/2/product"
            >
              <img
                alt="Product 2"
                class="border-2 border-black w-full h-60 object-cover"
                data-nimg="1"
                decoding="async"
                height="500"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=1080&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage2.jpg&w=1080&q=75 2x"
                style="color: transparent;"
                width="500"
              />
              <div
                class="flex flex-col font-extrabold uppercase p-1"
              >
                <h1>
                  Product 2
                </h1>
                <div
                  class="text-right"
                >
                  <p
                    class="line-through"
                  >
                    200
                     €
                  </p>
                  <p
                    class="text-red-600 text-lg"
                  >
                    20
                     €
                  </p>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
